### PR TITLE
Add property enable full list

### PR DESCRIPTION
**Pull request by wujack778**
Add property enable-full-list
[feature enable-full-list](https://github.com/Azure/azure-sdk-for-java/issues/20218)

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/GraphClient.java [hunk 1]
```diff
@@ -50,11 +50,13 @@ public GraphClient(AADAuthenticationProperties properties) {
                 LOGGER.error("Can not get group information from graph server.", ioException);
                 break;
             }
-            memberships.getValue()
-                       .stream()
-                       .filter(this::isGroupObject)
-                       .map(Membership::getDisplayName)
-                       .forEach(groups::add);
+            if(!properties.getUserGroup().getEnableFullList()){
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/GraphClient.java]: will it  have  npe here ?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 2]
```diff
@@ -161,6 +167,14 @@ public void setAllowedGroupNames(List<String> allowedGroupNames) {
             this.allowedGroupNames = allowedGroupNames;
         }
 
+        public Boolean getEnableFullList() {
+            return enableFullList;
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:170]: how about  
```
return enableFullList == null ? false:true  
```

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 3]
```diff
@@ -350,14 +364,17 @@ public void setAuthorizationClients(Map<String, AuthorizationClientProperties> a
     }
 
     public boolean isAllowedGroup(String group) {
+        if (this.getUserGroup().getEnableFullList()) {
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java]: I don't think we shoud add the code here

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 4]
```diff
@@ -411,7 +428,8 @@ public void afterPropertiesSet() throws Exception {
                             .filter(type -> !AADAuthorizationGrantType.AUTHORIZATION_CODE.equals(type))
                             .findAny()
                             .ifPresent(notUsed -> {
-                                throw new IllegalStateException("onDemand only support authorization_code grant type. ");
+                                throw new IllegalStateException("onDemand only support authorization_code grant type."
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java]: why change this

File: sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADAccessTokenGroupRolesExtractionTest.java [hunk 5]
```diff
@@ -123,11 +133,13 @@ public void testGroupsNameAndGroupsId() {
                 } catch (IOException ioException) {
                     break;
                 }
-                memberships.getValue()
-                           .stream()
-                           .filter(this::isGroupObject)
-                           .map(Membership::getDisplayName)
-                           .forEach(groups::add);
+                if(!properties.getUserGroup().getEnableFullList()){
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADAccessTokenGroupRolesExtractionTest.java]: @wujack778  , @ZhuXiaoBing-cn  will refactor the code here

File: sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADAccessTokenGroupRolesExtractionTest.java [hunk 6]
```diff
@@ -105,6 +106,15 @@ public void testGroupsNameAndGroupsId() {
         assertThat(groupsName).hasSize(2);
     }
 
+    @Test
+    public void testEnableFullList(){
+        setup();
+        when(properties.getUserGroup().getEnableFullList()).thenReturn(true);
+        Set<String> groupId= userService.extractGroupRolesFromAccessToken(accessToken);
+        assertThat(groupId).hasSize(2);
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AADAccessTokenGroupRolesExtractionTest.java]: I think we should have two tests
one for getEnableFullList==true, and another for getEnableFullList==false

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml [hunk 7]
```diff
@@ -6,6 +6,7 @@ azure:
     user-group:
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml:6]: I don't think you should push to ZhuXiaoBing-cn:master
image

**Review by backwind1233 — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml (hunk 7 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-webapp/src/main/resources/application.yml:6]: get it

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java (hunk 3 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java]: If remove this, when enable-full-list is true, all groups will not be listed.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/GraphClient.java (hunk 1 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/GraphClient.java]: The default value of enableFullList is false

**Comment by wujack778** [comment]
[New pull request](https://github.com/Azure/azure-sdk-for-java/pull/21080) has been created.

**Pull request closed without merging**